File: go/clock/clock.go
// Package clock implements a date-less clock.
package clock

import "fmt"

// Clock represents a time-of-day.
type Clock int

const day = 24 * 60

// New returns a new Clock.
func New(hour, minute int) Clock {
	return Clock((((hour*60 + minute) % day) + day) % day)
}

// String converts a Clock to a string.
func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c/60, c%60)
}

// Add adds time to a Clock
func (c Clock) Add(minutes int) Clock {
	return New(0, int(c)+minutes)
}

// Subtract subtracts time from a Clock
func (c Clock) Subtract(minutes int) Clock {
	return New(0, int(c)-minutes)
}

File: python/clock/clock.py
DAY = 24 * 60

class Clock:
  def __init__(self, hour, minute):
    self._min = (hour * 60 + minute) % DAY

  def __repr__(self):
    return '%02d:%02d' % (self._min // 60, self._min % 60)

  def __eq__(self, other):
    return self._min == other._min

  def __add__(self, minutes):
    return Clock(0, self._min + minutes)

  def __sub__(self, minutes):
    return Clock(0, self._min - minutes)


# vim:ts=2:sw=2:expandtab
